hal/armv7m/imxrt/117x/imxrt.c

```diff
@@ -498,6 +498,26 @@ int _imxrt_setDevClock(int clock, int div, int mux, int mfd, int mfn, int state)
 }
 
 
+int _imxrt_setDirectLPCG(int clock, int on)
+{
+	volatile u32 *reg;
+
+	if (clock < pctl_lpcg_m7 || clock > pctl_lpcg_uniq_edt_i)
+		return -1;
+
+	reg = imxrt_common.ccm + 0x1800 + clock * 0x8;
```

Comment:
Is it possible to replace `0x1800` by enum value from `pctl_lpcg_*`?

Reply:
I get it, but `pctl_lpcg_*`  are clock gates so it should not utilize this enumeration set. The `0x1800` is a relative offset in `ccm` register (in documentation  `ccm_register + 6000h + a * 20h`), I can introduce additional `enum` for ccm register which will describe `ccm` sub-functions like `root clocks`, `lpcg` and others.

Reply:
Let's leave it as is for now, `imxrt1176` docs complicates `ccm` offsets that they cannot be nicely enumerated (using `enum`s).